pkg/resourcecollector/resourcecollector.go

```diff
@@ -1434,7 +1434,7 @@ func (r *ResourceCollector) mergeAndUpdateApplicationResource(
 	dynamicClient dynamic.ResourceInterface,
 	opts *Options) error {
 
-	if len(opts.RancherProjectMappings) == 0 {
+	if (opts != nil) && len(opts.RancherProjectMappings) == 0 {
```

Comment:
This isn't right. If opts == nil we are passing it in prepareRancherApplicationResource in line 1453 while the intent from what I understand is to return. It should be 
If (opts == nil) || len(opts.RancherProjectMappings) == 0{
return nil
}

Reply:
@pp511  yeah you are right, with || the short circuit evaluation will not make the second condition to be evaluated at all so we will not land in a nil pointer dereference. will change it.